flink-libraries/flink-table/src/test/java/org/apache/flink/table/api/java/stream/sql/SqlITCase.java

```diff
@@ -30,11 +30,125 @@
 import org.apache.flink.streaming.util.StreamingMultipleProgramsTestBase;
 import org.apache.flink.table.api.java.stream.utils.StreamTestData;
 import org.junit.Test;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class SqlITCase extends StreamingMultipleProgramsTestBase {
+	
+	@Test
+	public void testRowRegister() throws Exception {
```

Comment:
This feature should not be tested with (so many) expensive integration tests.
Integration tests add a lot to the long build time and should be avoided if the feature can be tested with a more specific unit test. The `TableEnvironmentTest` has already many tests for `getFieldInfo`. We should add tests for `Row` there as well.